cnudie/util.py

```diff
@@ -414,6 +417,140 @@ def find_changed_component(
     )
 
 
+def format_component_diff(
+    component_diff: ComponentDiff,
+    delivery_dashboard_url_view_diff: str | None=None,
+    delivery_dashboard_url: str | None=None
+):
+    if delivery_dashboard_url_view_diff:
+        bom_diff_header = f'## <a href="{delivery_dashboard_url_view_diff}">BoM Diff</a>\n'
+    else:
+        bom_diff_header = '## BoM Diff\n'
+
+    added_components = [
+        f'\U00002795 {component.name} {component.version}'
+        for component in component_diff.cidentities_only_right
+        if component.name not in component_diff.names_version_changed
+    ]
+
+    removed_components = [
+        f'\U00002796 {component.name} {component.version}'
+        for component in component_diff.cidentities_only_left
+        if component.name not in component_diff.names_version_changed
+    ]
+
+    changed_components = [
+        f'\U00002699 {new_component.name}: {old_component.version} → {new_component.version}'
+        for old_component, new_component in component_diff.cpairs_version_changed
+    ]
+
+    summary_counts = textwrap.dedent(f'''\
+        Added components: {len(added_components)}
+        Changed components: {len(changed_components)}
+        Removed components: {len(removed_components)}\n
+    ''')
+
+    summary_details = []
+    if added_components:
+        summary_details.append('### Added Components:\n' + '\n'.join(added_components) + '\n')
+
+    if removed_components:
+        summary_details.append('### Removed Components:\n' + '\n'.join(removed_components) + '\n')
+
+    if changed_components:
+        summary_details.append('### Changed Components:\n' + '\n'.join(changed_components) + '\n')
+
+    component_details = []
+    for old_component, new_component in component_diff.cpairs_version_changed:
+        component_name = f'{new_component.name}'
```

Comment:
nit: I think this variable is rather unnecessary, you can simply use `new_component.name` directly in the following conditional block (also it would not be required to explicitly parse it to a string here again).